client/src/includes/contentMetrics.ts

```diff
@@ -0,0 +1,86 @@
+interface ContentMetrics {
+  wordCount: number;
+  readingTime: number;
+}
+
+interface SegmentData {
+  segment: string;
+  isWordLike?: boolean | undefined;
+}
+
+export const getContentMetrics = (
+  lang: string,
+  text: string,
+): ContentMetrics => {
+  let wordCount = 0;
+
+  if (typeof Intl.Segmenter === 'function') {
+    const segmenter = new Intl.Segmenter(lang, { granularity: 'word' });
+    const segments: SegmentData[] = Array.from(segmenter.segment(text));
+    wordCount = segments.reduce(
+      (count, segment) => (segment.isWordLike ? count + 1 : count),
+      0,
+    );
+  } else {
+    // Fallback to regex if Intl.Segmenter is not supported
+    wordCount =
```

Comment:
To simplify, I’d prefer we leave word count and reading time blank if `Intl.Segmenter` isn’t available. My understanding is only one of our supported browsers doesn’t have it (Firefox ESR 115). The next Firefox ESR will be released in October and will definitely have `Intl.Segmenter`, so this is a very temporary situation.